tasks/new_e2e_tests.py

```diff
@@ -356,20 +354,66 @@ class TooManyLogsError(Exception):
     pass
 
 
-def pretty_print_logs(logs_per_test, max_size=250000):
+def pretty_print_test_logs(logs_per_test: list[tuple[str, str, str]], max_size):
     # Compute size in bytes of what we are about to print. If it exceeds max_size, we skip printing because it will make the Gitlab logs almost completely collapsed.
     # By default Gitlab has a limit of 500KB per job log, so we want to avoid printing too much.
     size = 0
-    for _, tests in logs_per_test.items():
-        for _, logs in tests.items():
-            size += len("".join(logs).encode())
+    for _, _, logs in logs_per_test:
+        size += len("".join(logs).encode())
     if size > max_size and running_in_ci():
         raise TooManyLogsError
-    for package, tests in logs_per_test.items():
-        for test, logs in tests.items():
-            with gitlab_section("Complete logs for " + package + "." + test, collapsed=True):
-                print("Complete logs for " + package + "." + test)
-                print("".join(logs))
+    for package, test, logs in logs_per_test:
+        with gitlab_section("Complete logs for " + package + "." + test, collapsed=True):
+            print("Complete logs for " + package + "." + test)
+            print("".join(logs))
+
+    return size
+
+
+def pretty_print_logs(result_json_path, logs_per_test, max_size=250000, flakes_file="flakes.yaml"):
+    """Pretty prints logs with a specific order.
+
+    Print order:
+        1. Failing and non flaky tests
+        2. Failing and flaky tests
+        3. Successful and non flaky tests
+        4. Successful and flaky tests
+    """
+
+    result_json_name = result_json_path.split("/")[-1]
+    result_json_dir = result_json_path.removesuffix('/' + result_json_name)
+    washer = TestWasher(test_output_json_file=result_json_name, flakes_file_path=flakes_file)
+    failing_tests, marked_flaky_tests = washer.parse_test_results(result_json_dir)
+    all_known_flakes = washer.merge_known_flakes(marked_flaky_tests)
+
+    try:
+        # (failing, flaky) -> [(package, test_name, logs)]
+        categorized_logs = defaultdict(list)
+
+        # Split flaky / non flaky tests
+        for package, tests in logs_per_test.items():
+            package_flaky = all_known_flakes.get(package, set())
+            package_failing = failing_tests.get(package, set())
+            for test_name, logs in tests.items():
+                categorized_logs[test_name in package_failing, test_name in package_flaky].append(
+                    (package, test_name, logs)
+                )
+
+        for failing, flaky in [(True, False), (True, True), (False, False), (False, True)]:
```

Comment:
```suggestion
        for result_category in [cat.failed, cat.flaky_failed, cat.success, cat.flaky_success]:
```
Something like this, wdyt?